library/L2_HAL/sensors/environment/light/temt6000x01.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "L2_HAL/sensors/environment/light.hpp"
+
+#include "L1_Peripheral/adc.hpp"
+#include "utility/log.hpp"
+
+namespace sjsu
+{
+class Temt6000x01 final : public Light
+{
+ public:
+  /// Pull-down resistor
+  static constexpr uint32_t kResistance = 10'000;
+  /// 12-bit ADC resoltion = 2^12 - 1 = 4'095
+  static constexpr uint32_t kResolution = 4'095;
+  /// ADC Reference voltage.
+  static constexpr units::voltage::microvolt_t kReferenceVoltage = 3.3_V;
+  /// The conversion ratio to convert from the ADC output to
+  /// illuminance.
+  ///
+  /// From Fig. 3 - Photo Current vs. Illuminance of TEMT6000X01 datasheet,
+  /// lux = 2 * I, where current, I, is in micro-amps
+  ///     = 2 * adc_out * ((Vref / resolution) / resistance)
+  static constexpr float kConversionRatio =
```

Comment:
The ADC -> Lux equation is not linear. If you check the datasheet you will see that the graph is logarithmic. We must convert the graph in the datasheet to an equation and use that.